### DISCUSSION TITLE
Is there anything special about iina's audio output?

### DISCUSSION THREAD

**Discussion by rainlycoris** [iina/iina#5163] [General]
I encountered a very strange phenomenon, when use macbook pro 14 M1 with electric wire, while using headphones connected to the 3.5mm port, iina would produce electric current sound (background noise) when playing video, but not when using spotify or other playback audio, or not playing any audio.

When I unplugged the power cord, there was no current sound.  Strangely, touching the metal case with my hand didn't improve.

When I changed to AVFoundation or switched the audio sample rate, there was no improvement, nor did I switch the volume, even I set the iina volume to 0.

I don't understand why there is no electric current sound when spotify and no audio are playing. Is there a ghost or something here?

The main oddity is why spotify has no electric current sound at all, while iina is particularly serious.  As for why this is happening, just say it's a psychic phenomenon. I've solved the problem by connecting the power to the left.

I'm not sure if I'm the only one with this problem, the problem may be entirely mine, but I still want to ask. Thanks.

**Comment by krackers** [comment]
IINA uses libmpv, which in turns uses CoreAudio and AVFoundation as you noted. Are you able to reproduce this with upstream mpv?

It doesn't use CoreAudio in any non-standard way, and the fact that the issue persisted with AVFoundation likely means it's not specifically an AO issue. But could you try some of the related options like `--ao=coreaudio_exclusive`  (to use coreaudio in exclusive mode), `--coreaudio-change-physical-format=yes` (to set the physical format of the device) [note before doing this you should open MIDI settings and note down your existing values. Mpv does try to restore the values, but it may not always be done properly].

**Reply by rainlycoris** [reply]
Mpv also has this problem, and I found that iina play audio does not have this problem. `--ao=coreaudio_exclusive` `--coreaudio-change-physical-format=yes` they worked well but the problem was not solved, I tried a few other options and they didn't work.

**Reply by krackers** [reply]
>iina play audio does not have this problem

What happens if you switch to "music mode" when playing video? Also what happens if you bump `--audio-buffer` to something like `1` or `1.5`.

I remember something about how the last generation of Intel macbooks had different behavior when charging on left vs right side, something about overheating or throttling if I remember. I wonder if something about displaying video combined with charging on one side might cause audio thread (or coreaudiod itself) to get less CPU time causing some intermittent issues. I remember there being issues when the M1 macs first came about about high CPU or memory usage causing audio crackling.

It's definitely a very interesting issue, I wonder if anyone else with 14" M1 mac can replicate. My other hypothesis is that somehow it is a hardware defect on your particular machine where there is some electrical noise leakage somehow messing with the DAC. Testing with USB headphones might help rule that out.

**Reply by rainlycoris** [reply]
The problem of switching to music mode still exists, but music mode is also playing video. Setting '--audio-buffer' is also useless. I don't have USB headphones, sorry, so I can't test right now.

But I found that ***turning off the video track*** would solve the problem.

The issue of audio and video synchronization is also questionable, I tried the `--video-sync=` but it didn't work.

**Reply by rainlycoris** [reply]
In addition, the MBP 14 M1 is powered directly when it is used, and if the wire is charging the battery at the same time, there will be no problem.

I also replaced the cable and adapter, but it didn't matter.

**Reply by rainlycoris** [reply]
I think the problem may be close to your guess. When I closed the audio track just play video only on iina and played it on spotify at the same time, the sound of electricity also appeared. Just turning off the iina audio track instead of muting is no electric current sound. 

Firefox also has this problem, but it seems a little quieter. Quicktime is completely fine.

**Reply by krackers** [reply]
Then it seems the likely culprit is something about CPU or GPU usage interfering with coreaudiod, especially since Quicktime is unaffected. Does it make a difference if you play 360p or 4k video? 

Also which version of osx are you on?

**Reply by rainlycoris** [reply]
The 360p or 4k problem still exists, no significant difference in electric current sound.

sonoma 14.6.1

**Reply by rainlycoris** [reply]
When I was running the 3d game on the virtual machine, there was a loud electric current sound, maybe this should confirm the conjecture even more.

**Reply by krackers** [reply]
Perhaps some hardware defect related to lack of shielding or something interfering with DAC, that is exacerbated with heavy GPU use? I have not seen any reports of this online, so I don't think it's likely to be a software issue. If it's an issue with only your machine you might consider a hw replacement.

**Reply by rainlycoris** [reply]
The increase in power consumption caused by the heavy use of the gpu, let the current interfere more, and assuming that the battery is recharged and the circuit is different from the direct power supply, the left side of the power supply is also different from the right side, is indeed a good explanation for all phenomena.

Anyway, I really appreciate your help.
